packages/components/src/components/select/select.ts

```diff
@@ -629,53 +760,105 @@ export default class SdSelect extends SolidElement implements SolidFormControl {
+      helpText: this.hasSlotController.test('help-text')
+    };
+
+    const hasLabel = this.label ? true : !!slots['label'];
+    const hasHelpText = this.helpText ? true : !!slots['helpText'];
     const hasClearIcon = this.clearable && !this.disabled && this.value.length > 0;
-    const isPlaceholderVisible = this.placeholder && this.value.length === 0;
 
+    const isInvalid = this.hasAttribute('data-user-invalid') && this.required && !this.checkValidity();
+    const isValid = this.hasAttribute('data-user-valid') && this.required && this.checkValidity();
+
+    // Hierarchy of input states:
+    const selectState = this.disabled
+      ? 'disabled'
+      : this.hasFocus && isInvalid
+        ? 'activeInvalid'
+        : this.hasFocus && isValid
+          ? 'activeValid'
+          : this.hasFocus || this.open
+            ? 'active'
+            : isInvalid
+              ? 'invalid'
+              : isValid
+                ? 'valid'
+                : 'default';
+
+    // Conditional Styles
+    const cursorStyles = this.disabled ? 'cursor-not-allowed' : 'cursor-pointer';
+
+    const textSize = this.size === 'sm' ? 'text-sm' : 'text-base';
+    const textColor = {
+      disabled: 'text-neutral-500',
+      readonly: 'text-black',
+      activeInvalid: 'text-error',
+      activeValid: 'text-success',
+      active: 'text-black',
+      invalid: 'text-error',
+      valid: 'text-success',
+      default: 'text-black'
+    }[selectState];
+
+    const borderColor = {
+      disabled: 'border-neutral-500',
+      readonly: 'border-neutral-800',
+      activeInvalid: 'border-error border-2',
+      activeValid: 'border-success border-2',
+      active: 'border-primary border-2',
+      invalid: 'border-error',
+      valid: 'border-success',
+      default: 'border-neutral-800'
+    }[selectState];
+
+    const iconMarginLeft = { sm: 'ml-1', md: 'ml-2', lg: 'ml-2' }[this.size];
+    const iconSize = {
+      sm: 'text-base',
+      md: 'text-lg',
+      lg: 'text-xl'
+    }[this.size];
+
+    // Template
     return html`
       <div
         part="form-control"
-        class=${classMap({
-      'form-control': true,
-      'form-control--small': this.size === 'small',
-      'form-control--medium': this.size === 'medium',
-      'form-control--large': this.size === 'large',
-      'form-control--has-label': hasLabel,
-      'form-control--has-help-text': hasHelpText
-    })}
+        class=${cx('form-control relative text-left', cursorStyles, textColor, textSize, this.open && 'z-10')}
       >
         <label
           id="label"
           part="form-control-label"
-          class="form-control__label"
+          class=${hasLabel && 'inline-block mb-2'}
           aria-hidden=${hasLabel ? 'false' : 'true'}
           @click=${this.handleLabelClick}
         >
-          <slot name="label">${this.label}</slot>
+          <slot name="label">${this.label}${this.required ? ' *' : ''}</slot>
```

Comment:
We should not handle different logics here. Please double-check whether we incorporate the 'required (*)' in the CSS or HTML and make the change consistently. Currently, it is displayed in the CSS for sd-radio, sd-switch, and sd-checkbox and in HTML for sd-input and sd-textarea.